staging/src/k8s.io/apiserver/pkg/authentication/instrumentedauthenticator/metrics_test.go

```diff
@@ -0,0 +1,327 @@
+			defer authenticatedAttemptsCounter.Reset()
+
+			request, _ := http.NewRequestWithContext(
+				authenticator.WithAudiences(
+					context.Background(),
+					tt.apiAudience),
+				http.MethodPost,
+				"http://foo",
+				nil)
+			tt.request.AuthenticateRequest(request)
+			if err := testutil.GatherAndCompare(registry, strings.NewReader(tt.want), metrics...); err != nil {
+				t.Fatal(err)
+			}
+		})
+	}
+}
+
+func TestUnionAuthMetrics(t *testing.T) {
+	metrics := []string{
+		"authenticated_user_requests",
+		"authentication_attempts",
+	}
+
+	testCases := []struct {
+		desc        string
+		want        string
+		request     authenticator.Request
+		apiAudience authenticator.Audiences
+	}{
+		{
+			desc: "union auth - all authenticators fail",
+			request: union.New(
+				WrapRequest(
+					authenticator.RequestFunc(
+						func(_ *http.Request) (*authenticator.Response, bool, error) {
+							return &authenticator.Response{
+									User: &user.DefaultInfo{Name: "admin"},
+								},
+								false,
+								nil
+						}),
+					"instrumented-auth-01"),
+				WrapRequest(
+					authenticator.RequestFunc(
+						func(_ *http.Request) (*authenticator.Response, bool, error) {
+							return &authenticator.Response{
+									User: &user.DefaultInfo{Name: "admin"},
+								},
+								false,
+								nil
+						}),
+					"instrumented-auth-02"),
+			),
+			want: `
+        # HELP authentication_attempts [ALPHA] Counter of authenticated attempts.
+        # TYPE authentication_attempts counter
+        authentication_attempts{authenticator="instrumented-auth-01",result="failure"} 1
+        authentication_attempts{authenticator="instrumented-auth-02",result="failure"} 1
+				`,
+		},
+		{
+			desc: "union auth - first authenticator fails - second succeeds",
+			request: union.New(
+				WrapRequest(
+					authenticator.RequestFunc(
+						func(_ *http.Request) (*authenticator.Response, bool, error) {
+							return &authenticator.Response{
+									User: &user.DefaultInfo{Name: "admin"},
+								},
+								false,
+								nil
+						}),
+					"instrumented-auth-01"),
+				WrapRequest(
+					authenticator.RequestFunc(
+						func(_ *http.Request) (*authenticator.Response, bool, error) {
+							return &authenticator.Response{
+									User: &user.DefaultInfo{Name: "admin"},
+								},
+								true,
+								nil
+						}),
+					"instrumented-auth-02"),
+			),
+			want: `
+       # HELP authenticated_user_requests [ALPHA] Counter of authenticated requests broken out by username.
+        # TYPE authenticated_user_requests counter
+        authenticated_user_requests{authenticator="instrumented-auth-02",username="admin"} 1
+        # HELP authentication_attempts [ALPHA] Counter of authenticated attempts.
+        # TYPE authentication_attempts counter
+        authentication_attempts{authenticator="instrumented-auth-01",result="failure"} 1
+        authentication_attempts{authenticator="instrumented-auth-02",result="success"} 1
```

Comment:
I feel like this will just be super noisy for every form of auth that is early in the chain.

I would like to see how this looks for a real series of authenticators:

1. request header
1. x509
1. legacy SA
1. new SA
1. oidc (might be hard to set up)
1. token webhook (might be hard to set up)
1. anonymous

You would then pass an credential meant for each authenticator and see what the overall metrics look like.